civihr_employee_portal/updates/7044.php

```diff
@@ -12,26 +12,26 @@ function civihr_employee_portal_update_7044() {
 
   $addresses = _get_personal_address_count();
   foreach ($addresses as $address) {
-    if ($address['total'] < 2) {
+    if ($address['total'] != 2) {
       continue;
     }
 
     $result = civicrm_api3('Address', 'get', [
       'location_type_id' => 'Personal',
-      'contact_id' => $address['contact_id'],
+      'contact_id' => $address['contact_id']
     ]);
-    $updated = 0;
-    foreach ($result['values'] as $index => $contactAddress) {
-      if ($updated === 0) {
-        $contactAddress['is_primary'] = 1;
+    $processed = 0;
+    foreach ($result['values'] as $contactAddress) {
+      $updateAnyAddress = $contactAddress['is_primary'] == 0 && $address['primary_count'] == 1;
+      $updateSecondAddress = $address['primary_count'] != 1 && $processed == 1;
```

Comment:
This test for the scenario below

|Address|is_primary|
|--|--|
|Personal Address 1| 1|
|Personal Address 2| 1|

OR

|Address|is_primary|
|--|--|
|Personal Address 1| 0|
|Personal Address 2| 0|

Updates any of the two address with is_primary `0` to Work Address